### PR TITLE
Adding Atom10Reader - Syndicationfeed

### PR DESCRIPTION
**Pull request by ElJerry**

### CODE REVIEW

File: lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedFormatter.cs [hunk 1]
```diff
@@ -233,7 +243,121 @@ private SyndicationLink ParseLink(XmlReader reader)
...
+
+            if (!isEmpty)
+            {
+                FillItems(item, reader);
+            }
+
+            return item;
+        }
+
+        private void FillItems(SyndicationItem item, XmlReader reader)
```
Comment (drago-draganov) [inline lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedFormatter.cs]: typo, should be FillItem

File: lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedFormatter.cs [hunk 2]
```diff
@@ -30,12 +32,55 @@ public ISyndicationLink ParseLink(string value)
 
         public ISyndicationPerson ParsePerson(string value)
```
Comment (drago-draganov) [inline lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedFormatter.cs]: These should be virtual functions.

File: lab/src/Microsoft.SyndicationFeed/src/Atom/AtomFeedFormatter.cs [hunk 3]
```diff
@@ -99,15 +109,15 @@ public bool TryParseValue<T>(string value, out T result)
...
-            //    if (DateTimeUtils.TryParse(value, out dt))
-            //    {
-            //        result = (T)(object)dt;
-            //        return true;
-            //    }
-            //}
+            if (type == typeof(DateTimeOffset))
+            {
+                DateTimeOffset dt;
+                if (DateTimeUtils.TryParseAtom(value, out dt))
```
Comment (drago-draganov) [inline lab/src/Microsoft.SyndicationFeed/src/Atom/AtomFeedFormatter.cs]: Remove these functions from DateTimeUtils and place them into their respective formatters. All utility classes should have only common logic, not specific for a given protocol.

File: lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedFormatter.cs [hunk 4]
```diff
@@ -30,12 +32,55 @@ public ISyndicationLink ParseLink(string value)
...
+            var person = new SyndicationPerson();
+
+            person.RelationshipType = reader.Name;
+
+            //Read inner elements of person
+            reader.ReadStartElement();
+
+            while (reader.IsStartElement())
+            {
+                switch (reader.LocalName)
```
Comment (drago-draganov) [inline lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedFormatter.cs]: Make sure that every switch statement has an explicit default block.

File: lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedReader.cs [hunk 5]
```diff
@@ -47,5 +59,22 @@ protected override SyndicationElementType MapElementType(string elementName)
                     return SyndicationElementType.Content;
             }
         }
+
+        private async Task InitRead()
+        {
+            // Check <feed>
+            bool knownFeed = _reader.IsStartElement(Atom10Constants.FeedTag, Atom10Constants.Atom10Namespace);
```
Comment (drago-draganov) [inline lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedReader.cs]: No need for temp variable

Comment (drago-draganov) [inline lab/src/Microsoft.SyndicationFeed/src/Atom10/Atom10FeedReader.cs]: Doesn't look like it benefits from a temp variable

File: lab/src/Microsoft.SyndicationFeed/src/Atom/AtomFeedReader.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            {
+                //Read <feed>
+                await XmlUtils.ReadAsync(_reader);
+            }
+            else
+            {
+                throw new XmlException("Unkown Atom Feed");
+            }
+        }
+
```
Comment (drago-draganov) [inline lab/src/Microsoft.SyndicationFeed/src/Atom/AtomFeedReader.cs]: Extra line

**Comment by ElJerry** [comment]
Once its good to merge, should we squash all the commits?

**Comment by drago-draganov** [comment]
Looks reasonable.

**Pull request merged**